src/mmds/src/token.rs

```diff
@@ -378,12 +378,15 @@ mod tests {
         // We allow a deviation of 20ms to account for the gap
         // between the two calls to `get_time_ms()`.
         let deviation = 20;
-        assert!(ttl >= MILLISECONDS_PER_SECOND - deviation && ttl <= MILLISECONDS_PER_SECOND);
+        assert!(
+            ttl >= MILLISECONDS_PER_SECOND - deviation && ttl <= MILLISECONDS_PER_SECOND,
+            "ttl={ttl} not within +-{deviation} of {MILLISECONDS_PER_SECOND}",
```

Comment:
This condition evaluates the `ttl` between `MILLISECONDS_PER_SECOND - deviation` and `MILLISECONDS_PER_SECOND`, not between `MILLISECONDS_PER_SECOND - deviation` and `MILLISECONDS_PER_SECOND + deviation`. So the message `+-{deviation}` looks not be appropriate or the condition itself might not be proper.

Reply:
You are right, good catch! `ttl` can only every be larger than 1 second, not smaller, so the assert allows deviation in exactly the wrong direction. I would assume that this is indeed the reason we were observing the spurious failures.